decidim-assemblies/app/views/decidim/assemblies/admin/assemblies/_form.html.erb

```diff
@@ -157,7 +157,7 @@
       <div class="row column">
         <%= form.select :parent_id,
                         options_from_collection_for_select(
-                          @all_assemblies,
+                          @all_assemblies.where.not(id: current_assembly),
```

Comment:
Could we please make the code better since we have to change it?

At the assemblies controller, instead of having a `before_action` that sets an instance variable, change it to a helper method:
```suggestion
                          parent_assemblies,
```

```ruby
helper_method :current_assembly, :parent_assembly, :parent_assemblies, :current_participatory_space, :query, :organization assemblies

# Change set_all_assemblies_method
def parent_assemblies
  @parent_assemblies ||= organization_assemblies.where.not(id: current_assembly)
end
```

Having instance variables in views could be considered a bad practice since the error message when they are nil is not really helpful.